From 992877a061a41be81f4f85973edb60041c943397 Mon Sep 17 00:00:00 2001
From: Franck LENORMAND <franck.lenormand@nxp.com>
Date: Mon, 10 Sep 2018 11:54:24 +0200
Subject: [PATCH 4621/5242] MLK-19434: crypto: testmgr: Add log printing once
 if disabled

commit  65fe444b43eb8eaf667e40326605a711439ffbb3 from
https://source.codeaurora.org/external/imx/linux-imx.git

If the test manager is not disable, it is not possible to
determine if tcrypt result is suitable or not.

This patch fix this issue printing a message to the user.

Signed-off-by: Franck LENORMAND <franck.lenormand@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 crypto/testmgr.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/crypto/testmgr.c b/crypto/testmgr.c
index 59b98c0..3f8e037 100644
--- a/crypto/testmgr.c
+++ b/crypto/testmgr.c
@@ -46,6 +46,8 @@
 /* a perfect nop */
 int alg_test(const char *driver, const char *alg, u32 type, u32 mask)
 {
+	printk_once(KERN_INFO "alg: self-tests disabled by "
+		"CONFIG_CRYPTO_MANAGER_DISABLE_TESTS\n");
 	return 0;
 }
 
-- 
1.7.9.5

